api/v1alpha1/jobset_types.go

```diff
@@ -71,9 +71,15 @@ type ReplicatedJob struct {
 	Template batchv1.JobTemplateSpec `json:"template,omitempty"`
 	// Network defines the networking options for the job.
 	Network *Network `json:"network"`
+	// Replicas is the number of jobs that will be created from this ReplicatedJob's template.
+	// Pod hostnames for pods created by these jobs will be in the format:
+	// <jobsetName>-<jobName>-<jobIndex>-<podIndex>
+	Replicas *int `json:"replicas,omitempty"`
 }
 type Network struct {
 	// EnableDNSHostnames allows pods to be reached via their hostnames.
+	// Pods will be reachable using the fully qualified pod hostname, which is in
```

Comment:
I'm thinking we should also mark the JobSet as failed if it fails to create a headless service for a job since it is not in indexed completion mode. Do you agree? 

I implemented this check in the reconciler, and added a test case for it as well, but haven't pushed the changes yet. Do you mind if we include this new validation and test case in a separate PR?

Reply:
sounds good